frontends/web/src/components/wallet-connect/incoming-signing-request-dialog.tsx

```diff
@@ -0,0 +1,145 @@
+  const { t } = useTranslation();
+  return (
+    <div className={styles.animationAndTextContainer}>
+      <AnimatedChecked className={styles.successIcon} />
+      <p>{t('walletConnect.signingRequest.successfullySigned')}</p>
+    </div>
+  );
+};
+
+
+export const WCIncomingSignRequestDialog = ({
+  open,
+  onAccept,
+  onReject,
+  content,
+  stage,
+}: TRequestDialogProps) => {
+  const { t } = useTranslation();
+  const { isDarkMode } = useDarkmode();
+  const { accountAddress, accountName, signingData, chain, method, currentSession } = content;
+
+  const formattedChain = chain in SUPPORTED_CHAINS ? SUPPORTED_CHAINS[chain].name : chain;
+  const chainIcon = chain in SUPPORTED_CHAINS ? SUPPORTED_CHAINS[chain].icon : null;
+
+  const metadata = currentSession.peer.metadata;
+  const dappHasIcon = metadata.icons && metadata.icons.length > 0;
+
+  return (
+    <Dialog open={open} large onClose={onReject}>
+      <div className={styles.titleContainer}>
+        {isDarkMode ? <WalletConnectLight height={32} width={32} /> : <WalletConnectDark height={40} width={40} />}
+        <h3>{t('walletConnect.signingRequest.walletConnectRequest')}</h3>
+      </div>
+      <div className={styles.outerContainer}>
+
+        {
+          stage !== 'accepted' && <>
+            <ul className={styles.listContainer}>
+              <div className={styles.iconContainer}>
+                <li className={styles.item}>
+                  <p className={styles.label}>{t('walletConnect.signingRequest.account')}</p>
+                  <span className={styles.accountNameAndAddress}>
+                    <p className={styles.accountName}><b>{accountName}</b></p>
+                    <p className={styles.address}>{truncateAddress(accountAddress)}</p>
+                  </span>
+                </li>
+                {dappHasIcon && <img className={styles.dappIcon} src={metadata.icons[0]} alt="icon" /> }
+              </div>
+
+              <li className={styles.item}>
+                <p className={styles.label}>{t('walletConnect.signingRequest.chain')}</p>
+                <div className={styles.chainContainer}>
+                  <p className={styles.itemText}>{formattedChain}</p>
+                  {chainIcon}
+                </div>
+              </li>
+
+              <li className={styles.item}>
+                <p className={styles.label}>{t('walletConnect.signingRequest.dapp')}</p>
+                <p className={styles.itemText}>{metadata.name}</p>
+              </li>
+
+              <li className={styles.item}>
+                <p className={styles.label}>{t('transaction.details.type')}</p>
+                <p className={styles.itemText}>{method}</p>
+              </li>
+
+              {signingData &&
+            (
+              <li className={styles.item}>
+                <p className={styles.label}>{t('walletConnect.signingRequest.data')}</p>
+                <textarea className={styles.textarea} readOnly value={signingData.toString()} />
```

Comment:
This can be a TODO for a later PR as well, but in the case of signing Typed Data, can we prettify the data field so it has JSON formatting (new lines and indents) so it's more readable and expand the input field + modal so the entire text is visible?

Reply:
does it have complicated nested data? I guess `JSON.stringify(data, null, 2);` could work.

Reply:
for me, whatever works to make it a bit more readable and not too scary to the average user